ruby/basics/loops.md

```diff
@@ -17,35 +20,111 @@ what the student is expected to know or be able to do by the end of this lesson
 * You know how to use the `upto` and `downto` loops in ruby
 
 ### Loop
-* Rubys loop loop lol
-* what its good for
-* how to break out of it
-* what its good for
+The loop loop (say what????) is Ruby's loop that just don't quit. It's an infinite loop that'll keep going unless you specifically request for it to stop using the `break` command. Break is commanly used with a condition as illustrated in the code example below.
+
+```ruby
+i = 0
+loop do
+  puts "i is #{i}"
+  i += 1
+  break if i == 10
+end
+```
+
+You won't see this loop used much in Ruby and if you're using it then there is probably a better loop for you out there.
 
 ### While Loop
-* how it works
-* include code snippets for a while loop that loops a certain amount of times and one that loops until a condition is met.
+
+A while loop is similar to the `loop` loop except you declare the condition that will break out of the loop up front.
+
+```ruby
+i = 0
+while(i < 10) do
+ puts "i is #{i}"
+ i += 1
+end
+```
+This is an example of using a while loop with a count, and as you declare the condition that breaks the loop up-front, the intention of your code is much clearer making it easier to read than our loop loop above.
+
+While loops can also be used to repeatedly ask a question of the user until they give an appropriate response.
+
+```ruby
+while gets.chomp != "yes" do
+  puts "Will you go to prom with me?"
+end
+```
+In real life you obviously should just take no for an answer the first time.
+
 
 ### For Loop
-* how it works
-* a few code snippets
-* mention its barely used in favour of enumerables
+
+For loops are used to iterate through a collection of information such as an array or range.
```

Comment:
Might want to wrap "For" in backticks, again for consistency and differentiation, as it's such a common English word.